### PR TITLE
Work around changes in xarray 0.15

### PR DESCRIPTION
**Pull request by mraspaud**
Xarray v0.15.0 brings some changes that broke the satpy unit tests.
Here we provide a fix for coarsen and disallow xarray 0.15.0 because of a bug: 
https://github.com/pydata/xarray/issues/3746

 - [x] Tests passed 
 - [x] Passes ``flake8 satpy``

### CODE REVIEW

File: satpy/scene.py [hunk 1]
```diff
@@ -665,7 +665,7 @@ def aggregate(self, dataset_ids=None, boundary='exact', side='left', func='mean'
             except AttributeError:
                 resolution = max(target_area.lats.resolution, target_area.lons.resolution)
             for ds_id in ds_ids:
-                res = self[ds_id].coarsen(boundary=boundary, side=side, func=func, **dim_kwargs)
+                res = self[ds_id].coarsen(boundary=boundary, side=side, **dim_kwargs)
```
Comment (djhoese) [inline satpy/scene.py]: Don't you need to pass the `func` positional argument?

Comment (mraspaud) [reply satpy/scene.py]: no, it isn't needed. It was actually `coord_func` and I don't think it needs to be the same as `func`, so I'm dropping it.

Comment (djhoese) [reply satpy/scene.py]: Ah I didn't see it is used below.

**Comment by mraspaud** [comment]
I thinks this should be squash-merged

**Comment by djhoese** [comment]
If this passes tests, is it ready for merge?

Based on the changes, I assume you searched for `del ...coords['coord']` usage?

**Comment by mraspaud** [comment]
Yes, and yes.

**Pull request merged**